core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -423,17 +436,19 @@ impl SchedulerController {
         let remaining_queue_capacity = self.container.remaining_queue_capacity();
 
         const MAX_PACKET_RECEIVE_TIME: Duration = Duration::from_millis(100);
-        let recv_timeout = match decision {
-            BufferedPacketsDecision::Consume(_) => {
+        let (recv_timeout, should_buffer) = match decision {
+            BufferedPacketsDecision::Consume(_) => (
                 if self.container.is_empty() {
                     MAX_PACKET_RECEIVE_TIME
                 } else {
                     Duration::ZERO
-                }
+                },
+                true,
+            ),
+            BufferedPacketsDecision::Forward => (MAX_PACKET_RECEIVE_TIME, false),
```

Comment:
Hmmmm how does this work?

If I'm understanding, we will no longer buffer packets when the decision is to forward (i.e. we're not leader and don't have a leader slot coming up). This makes sense for the default "don't forward" behavior. But don't we want to buffer when we have staked overrides set?

Reply:
Yeah you're right...probably not caught because the only tests we have for forwarding have so few validators they are always either leader or within 20 slots of being leader.

Reply:
This should be equivalent to `forwarding_enabled`; when forwarding is disabled we don't buffer - if it is we buffer and forward.

Reply:
#2305 fix for master